tools/benchmarksAnalyzer/src/main/kotlin/main.kt

```diff
@@ -15,49 +15,110 @@
-            OptionDescriptor(ArgType.DOUBLE, "eps", "e", "Meaningful performance changes", "0.5"),
-            OptionDescriptor(ArgType.BOOLEAN, "short", "s", "Show short version of report", "false")
+            OptionDescriptor(ArgType.String(), "output", "o", "Output file"),
+            OptionDescriptor(ArgType.Double(), "eps", "e", "Meaningful performance changes", "0.5"),
+            OptionDescriptor(ArgType.Boolean(), "short", "s", "Show short version of report", "false"),
+            OptionDescriptor(ArgType.Choice(listOf("text", "html", "teamcity")),
+                    "renders", "r", "Renders for showing information", "text", isMultiple = true),
+            OptionDescriptor(ArgType.String(), "user", "u", "User access information for authorization")
     )
 
     val arguments = listOf(
-            ArgDescriptor(ArgType.STRING, "mainReport", "Main report for analysis"),
-            ArgDescriptor(ArgType.STRING, "compareToReport", "Report to compare to", isRequired = false)
+            ArgDescriptor(ArgType.String(), "mainReport", "Main report for analysis"),
+            ArgDescriptor(ArgType.String(), "compareToReport", "Report to compare to", isRequired = false)
     )
 
     // Parse args.
     val argParser = ArgParser(options, arguments)
     if (argParser.parse(args)) {
         // Read contents of file.
-        val mainBenchsResults = readFile(argParser.get("mainReport")!!.stringValue)
+        val mainBenchsResults = getFileContent(argParser.get<String>("mainReport")!!, argParser.get<String>("user"))
         val mainReportElement = JsonTreeParser.parse(mainBenchsResults)
         val mainBenchsReport = BenchmarksReport.create(mainReportElement)
-        var compareToBenchsReport = argParser.get("compareToReport")?.stringValue?.let {
-            val compareToResults = readFile(it)
+        var compareToBenchsReport = argParser.get<String>("compareToReport")?.let {
+            val compareToResults = getFileContent(it, argParser.get<String>("user"))
             val compareToReportElement = JsonTreeParser.parse(compareToResults)
             BenchmarksReport.create(compareToReportElement)
         }
 
-        // Generate comparasion report
+        val renders = argParser.getAll<String>("renders")
+
+        // Generate comparasion report.
         val summaryReport = SummaryBenchmarksReport(mainBenchsReport,
                 compareToBenchsReport,
-                argParser.get("eps")!!.doubleValue)
-        TextRender().print(summaryReport, argParser.get("short")!!.booleanValue,
-                argParser.get("output")?.stringValue)
-        // Produce information for TeamCity if needed.
-        getEnv("TEAMCITY_BUILD_PROPERTIES_FILE")?.let {
-            TeamCityStatisticsRender().print(summaryReport, argParser.get("short")!!.booleanValue)
+                argParser.get<Double>("eps")!!)
+
+        var output = argParser.get<String>("output")
+
+        renders?.forEach {
```

Comment:
What happens if user specifies several renders? Does the output file contain several reports in different formats in this case?